code/drasil-gool/lib/GOOL/Drasil/LanguageRenderer/PythonRenderer.hs

```diff
@@ -1032,3 +1032,10 @@ pyDocComment :: [String] -> Doc -> Doc -> Doc
 pyDocComment [] _ _ = empty
 pyDocComment (l:lns) start mid = vcat $ start <+> text l : map ((<+>) mid . 
   text) lns
+
+toConstName :: String -> String
+toConstName (s:s':ss) = if isLower s && isUpper s'
+                          then toUpper s : '_' : toUpper s' : toConstName ss
```

Comment:
```suggestion
                          then toUpper s : '_' : s' : toConstName ss
```

Super minor refactoring that should work (😅), since we know that `isUpper s' == True`

Not sure if this was intentional, but as far as I can tell, this function would format both `myId` _and_ `myID` as `MY_ID`, which I think is ideal!

A nice next step would be handling strings of all caps in the input, but then the user input of either `inputHTMLcode` or `inputHTMLCode` would be ambiguous, so I think what you have is sufficient for now!

Reply:
Good catch!

Yeah true, I guess there'll always be a point where smart-renaming fails.  We _could_ do something like look for two capitals followed by a lowercase, and add an `_` between the two capitals.  That would work with `inputHTMLCode` (which seems more conventional to me than `inputHTMLcode`), but that might be a bit too 'clever' and lead to unexpected variable names under some conditions.